src/Mobs/Spider.cpp

```diff
@@ -33,3 +34,37 @@ void cSpider::GetDrops(cItems & a_Drops, cEntity * a_Killer)
 
 
 
+
+void cSpider::EventSeePlayer(cEntity * a_Entity)
+{
+	if (!static_cast<cPlayer *>(a_Entity)->IsGameModeCreative() && (this->GetWorld()->GetBlockBlockLight(this->GetPosition()) <= 9))
```

Comment:
For consistency with the rest of the code, perhaps remove the `this` keyword?

I suspect that before you get the block light, you will also need to check cWorld::IsChunkLighted and call cWorld::QueueLightChunk and abort if it isn't. You would also have to get the current chunk position through cChunkDef::BlockToChunk(POSX_TOINT, ...).